### PR TITLE
Feat refactor api

### PR DESCRIPTION
**Pull request by francoisbonneau**

### CODE REVIEW

File: include/geode/inspector/information.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+#include <string>
+#include <vector>
+
+#include <geode/inspector/common.h>
+
+namespace geode
+{
+    template < typename ProblemType >
+    struct ProblemInspectionResult
```
Comment (MelchiorSchuh) [inline include/geode/inspector/information.h:34]: renommer en InspectionIssues  ?

File: include/geode/inspector/topology/brep_lines_topology.h [hunk 2]
```diff
@@ -22,16 +22,49 @@
...
+                "Indices of unique vertices linked to a line with invalid "
+                "internal topology."
+            };
+        ProblemInspectionResult< index_t >
+            unique_vertices_linked_to_a_single_and_invalid_line{
+                "Indices of unique vertices linked to only one line and this "
+                "single line is invalid."
+            };
+        ProblemInspectionResult< index_t >
+            unique_vertices_linked_to_a_line_but_not_linked_to_a_corner{
```
Comment (MelchiorSchuh) [inline include/geode/inspector/topology/brep_lines_topology.h:62]:
```suggestion
            unique_vertices_linked_to_several_lines_but_not_linked_to_a_corner{
```

File: include/geode/inspector/topology/brep_lines_topology.h [hunk 3]
```diff
@@ -22,16 +22,49 @@
...
+                "Indices of unique vertices linked to a line but not kinked to "
+                "a "
+                "corner."
```
Comment (MelchiorSchuh) [inline include/geode/inspector/topology/brep_lines_topology.h:65]:
```suggestion
                "Indices of unique vertices linked to several lines but not linked to "
                "a corner."
```

File: include/geode/inspector/topology/brep_surfaces_topology.h [hunk 4]
```diff
@@ -22,16 +22,55 @@
...
+        ProblemInspectionResult< index_t >
+            unique_vertices_linked_to_a_surface_with_invalid_embbedings{
+                "Indices of unique vertices linked to a surface with invalid "
+                "internal topology."
+            };
+        ProblemInspectionResult< index_t >
+            unique_vertices_linked_to_a_single_and_invalid_surface{
+                "Indices of unique vertices linked to only one surface and "
+                "this "
+                "single surface is invalid."
```
Comment (MelchiorSchuh) [inline include/geode/inspector/topology/brep_surfaces_topology.h:60]:
```suggestion
                "this single surface is invalid."
```

Comment (MelchiorSchuh) [reply include/geode/inspector/topology/brep_surfaces_topology.h:60]: Essaye d'enlever les lignes inutiles du meme genre dans le reste du code aussi

File: include/geode/inspector/topology/section_lines_topology.h [hunk 5]
```diff
@@ -21,17 +21,48 @@
...
+        ProblemInspectionResult< index_t >
+            unique_vertices_linked_to_a_single_and_invalid_line{
+                "Indices of unique vertices linked to only one line and this "
+                "single line is invalid."
+            };
+        ProblemInspectionResult< index_t >
+            unique_vertices_linked_to_a_line_but_not_linked_to_a_corner{
+                "Indices of unique vertices linked to a line but not kinked to "
+                "a "
+                "corner."
```
Comment (MelchiorSchuh) [inline include/geode/inspector/topology/section_lines_topology.h:63]:
```suggestion
            unique_vertices_linked_to_several_lines_but_not_linked_to_a_corner{
                "Indices of unique vertices linked to several lines but not linked to "
                "a corner."
```

File: src/geode/inspector/criterion/colocation/unique_vertices_colocation.cpp [hunk 6]
```diff
@@ -321,7 +321,7 @@ namespace geode
                     {
                         point_group_string += " ";
                         point_group_string +=
-                            std::to_string( point_group[point_index] );
+                            absl::StrCat( point_group[point_index] );
```
Comment (MelchiorSchuh) [inline src/geode/inspector/criterion/colocation/unique_vertices_colocation.cpp:324]: idem

File: src/geode/inspector/criterion/colocation/unique_vertices_colocation.cpp [hunk 7]
```diff
@@ -254,7 +254,7 @@ namespace geode
...
```
Comment (MelchiorSchuh) [inline src/geode/inspector/criterion/colocation/unique_vertices_colocation.cpp:257]: point_group_string += absl::StrCat( " ", point_group[point_index] );

**Review by MelchiorSchuh — COMMENTED** [review]
A few comments, change the StrCat calls like I explained on Slack, and its good for me

File: include/geode/inspector/information.h [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        index_t number()
+        {
+            return problems.size();
+        }
+
+        void add_problem(
+            const ProblemType& problem, const std::string& message )
+        {
+            problems.push_back( problem );
+            messages.push_back( message );
```
Comment (BotellaA) [inline include/geode/inspector/information.h:50]:
```suggestion
            ProblemType problem, std::string message )
        {
            problems.emplace_back( std::move( problem ) );
            messages.emplace_back( std::move( message ) );
```

File: include/geode/inspector/topology/brep_blocks_topology.h [hunk 9]
```diff
@@ -23,15 +23,42 @@
...
     class BRep;
+    struct uuid;
 } // namespace geode
 
 namespace geode
 {
+    struct opengeode_inspector_inspector_api BRepBlocksTopologyInspectionResult
+    {
+        ProblemInspectionResult< uuid > blocks_not_meshed{
+            "uuids of block without mesh."
```
Comment (BotellaA) [inline include/geode/inspector/topology/brep_blocks_topology.h:47]: block or Block? Should we be close to the C++ syntax or not?

File: include/geode/inspector/topology/brep_topology.h [hunk 10]
```diff
@@ -26,19 +26,31 @@
...
 
 namespace geode
 {
-    struct ComponentMeshVertex;
     class BRep;
 } // namespace geode
 
 namespace geode
 {
+    struct opengeode_inspector_inspector_api BRepTopologyInspectionResult
```
Comment (BotellaA) [inline include/geode/inspector/topology/brep_topology.h:42]:
```suggestion
    struct BRepTopologyInspectionResult
```
No need if nothing is implemented in the cpp file

Comment (BotellaA) [reply include/geode/inspector/topology/brep_topology.h:42]: Do the same for other struct

File: include/geode/inspector/topology/brep_topology.h
Comment (BotellaA) [inline include/geode/inspector/topology/brep_topology.h:1]: BRepTopologyInspector( const BRep& brep, bool verbose ); could be removed

Comment (BotellaA) [reply include/geode/inspector/topology/brep_topology.h:1]: same for Section I guess

**Review by BotellaA — APPROVED** [review]

File: include/geode/inspector/topology/brep_blocks_topology.h (hunk 9 above)
Comment (francoisbonneau) [reply include/geode/inspector/topology/brep_blocks_topology.h:47]: I think block is fine... but as yo want.

File: include/geode/inspector/topology/brep_topology.h
Comment (francoisbonneau) [reply include/geode/inspector/topology/brep_topology.h:1]: yes, I will remove every verbose constructor at the end.

**Pull request merged**